third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -650,6 +656,26 @@ func resourceSqlDatabaseInstanceCreate(d *schema.ResourceData, meta interface{})
 
 	d.Set("name", name)
 
+	// Before we create the instance, check if at least 1 service connection exists for private SQL Instances.
+	network := d.Get("settings.0.ip_configuration.0.private_network").(string)
+	if network != "" {
+		// Borrow some of the functions from resource_service_networking_connection.go
+		serviceNetworkingNetworkName, err := retrieveServiceNetworkingNetworkName(d, config, network)
+		if err != nil {
+			return err
+		}
+		response, err := config.clientServiceNetworking.Services.Connections.List("services/servicenetworking.googleapis.com").
+			Network(serviceNetworkingNetworkName).Do()
+		if err != nil {
+			// It is possible that the identity creating the SQL Instance might not have permissions to call servicenetworking.services.connections.list
+			log.Printf("[WARNING] Failed to list Service Networking of the project. Skipping Service Networking check.")
+		} else {
+			if len(response.Connections) < 1 {
+				return fmt.Errorf("Error, failed to create instance because the network doesn't have atleast 1 private services connections. Please review the docs on how to create this connection.")
```

Comment:
typo: "at least", and "connection".  Can you also link the docs?

Reply:
Fixed